src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonAddressService.kt

```diff
@@ -32,21 +34,51 @@ class PrisonAddressService(
       address.postcode = postcode
       address.country = country
     }
-    val trackingAttributes = mapOf(
-      "prisonId" to prisonId,
-      "addressId" to addressId.toString(),
-      "addressLine1" to updateAddressRecord.addressLine1,
-      "addressLine2" to updateAddressRecord.addressLine2,
-      "town" to updateAddressRecord.town,
-      "county" to updateAddressRecord.county,
-      "postcode" to updateAddressRecord.postcode,
-      "country" to updateAddressRecord.country,
-    )
-
-    telemetryClient.trackEvent("prison-register-address-update", trackingAttributes, null)
+
+    recordPrisonAddressEditEvent("prison-register-address-update", prisonId, addressId.toString(), updateAddressRecord)
     return AddressDto(address)
   }
 
+  @Transactional
+  fun addAddress(prisonId: String, additionalAddress: UpdateAddressDto): AddressDto {
+    val prison = prisonRepository.findById(prisonId)
+      .orElseThrow { EntityNotFoundException("Prison $prisonId not found") }
+
+    with(additionalAddress) {
+      val address = addressRepository.save(
+        Address(
+          addressLine1 = addressLine1,
+          addressLine2 = addressLine2,
+          town = town,
+          county = county,
+          postcode = postcode,
+          country = country,
+          prison = prison
+        )
+      )
+
+      recordPrisonAddressEditEvent("prison-register-add-address", prisonId, address.id?.toString(), additionalAddress)
```

Comment:
Not a deal breaker - but would probably have consistency in event name i.e. have both as prison-register-address-add/update or prison-register-add/update-address

Reply:
Hi Karen - agreed - updated for better consistency